src/ast/nodes/ConditionalExpression.ts

```diff
@@ -78,7 +78,19 @@ export default class ConditionalExpression extends NodeBase implements Deoptimiz
 		origin: DeoptimizableEntity
 	): LiteralValueOrUnknown {
 		const usedBranch = this.getUsedBranch();
-		if (!usedBranch) return UnknownValue;
+		if (!usedBranch) {
+			const consequentValue = this.consequent.getLiteralValueAtPath(path, recursionTracker, origin);
+			const castedConsequentValue = tryCastLiteralValueToBoolean(consequentValue);
+			if (castedConsequentValue === UnknownValue) return UnknownValue;
+			const alternateValue = this.alternate.getLiteralValueAtPath(path, recursionTracker, origin);
+			const castedAlternateValue = tryCastLiteralValueToBoolean(alternateValue);
+			if (castedAlternateValue === UnknownValue) return UnknownValue;
```

Comment:
I think we can skip this line as in that case, the values will not be the same in the next line.

Reply:
I took the liberty to push this small fix so this can be merged